includes/class-wc-payments-upe-checkout.php

```diff
@@ -88,6 +95,43 @@ public function __construct(
 		add_action( 'wc_ajax_wcpay_init_setup_intent', [ $this->gateway, 'init_setup_intent_ajax' ] );
 		add_action( 'wc_ajax_wcpay_log_payment_error', [ $this->gateway, 'log_payment_error_ajax' ] );
 
+		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts' ] );
+		add_action( 'wp_enqueue_scripts', [ $this, 'register_scripts_for_zero_order_total' ], 11 );
+	}
+
+	/**
+	 * Registers all scripts, necessary for the gateway.
+	 */
+	public function register_scripts() {
+		// Register Stripe's JavaScript using the same ID as the Stripe Gateway plugin. This prevents this JS being
+		// loaded twice in the event a site has both plugins enabled. We still run the risk of different plugins
+		// loading different versions however. If Stripe release a v4 of their JavaScript, we could consider
+		// changing the ID to stripe_v4. This would allow older plugins to keep using v3 while we used any new
+		// feature in v4. Stripe have allowed loading of 2 different versions of stripe.js in the past (
+		// https://stripe.com/docs/stripe-js/elements/migrating).
+		wp_register_script(
+			'stripe',
+			'https://js.stripe.com/v3/',
+			[],
+			'3.0',
+			true
+		);
+
+		$script_dependencies = [ 'stripe', 'wc-checkout', 'wp-i18n' ];
+
+		if ( $this->gateway->supports( 'tokenization' ) ) {
+			$script_dependencies[] = 'woocommerce-tokenization-form';
+		}
+
+		if ( ! WC_Payments_Features::is_upe_split_enabled() ) {
+			$script = 'dist/upe_checkout';
+		} elseif ( WC_Payments_Features::is_upe_deferred_intent_enabled() ) {
+			$script = 'dist/upe_with_deferred_intent_creation_checkout';
+		} else {
+			$script = 'dist/upe_split_checkout';
+		}
```

Comment:
```suggestion
		if ( WC_Payments_Features::is_upe_split_enabled() ) {
			$script = 'dist/upe_split_checkout';
		} elseif ( WC_Payments_Features::is_upe_deferred_intent_enabled() ) {
			$script = 'dist/upe_with_deferred_intent_creation_checkout';
		} else {
			$script = 'dist/upe_checkout';
		}
```
I've just noticed that when the new UPE with deferred intent is enabled, the checkout scripts do not get enqueued. This is because when the UPE split is not enabled, we will enqueue the legacy UPE scripts and ignore everything else. 

The small change above resolves this and I believe keeps everything else chugging along too. Let me know if that works for you!

![New UPE with deferred intent feature flag](https://cdn-std.droplr.net/files/acc_1104206/qXWawd)
_New UPE with deferred intent feature flag FYI_

Reply:
Perfect, that's actually a much preferred order. Nice one!